src/main/java/net/apnic/rdap/ip/IPStatsResourceLocator.java

```diff
@@ -46,7 +46,7 @@ public RDAPAuthority authorityForResource(IpRange range)
      * {@inheritDocs}
      */
     @Override
-    public void putResourceMapping(IpRange resource, RDAPAuthority authority)
+    public synchronized void putResourceMapping(IpRange resource, RDAPAuthority authority)
```

Comment:
(suggestion) having the scrapers return an (immutable) result that can be aggregated imo would be less error prone than concurrent modifications.